ray-operator/test/e2e/raycluster_test.go

```diff
@@ -177,16 +156,105 @@ func TestRayClusterGCSFT(t *testing.T) {
 	test.T().Logf("Created RayCluster %s/%s successfully", rayCluster.Namespace, rayCluster.Name)
 
 	// Make sure the RAY_REDIS_ADDRESS env is set on the Head Pod.
-	g.Eventually(func(g Gomega) bool {
-		rayCluster, err := test.Client().Ray().RayV1().RayClusters(namespace.Name).Apply(test.Ctx(), rayClusterAC, TestApplyOptions)
-		g.Expect(err).NotTo(HaveOccurred())
-		if rayCluster.Status.Head.PodName != "" {
-			headPod, err := test.Client().Core().CoreV1().Pods(namespace.Name).Get(test.Ctx(), rayCluster.Status.Head.PodName, metav1.GetOptions{})
-			g.Expect(err).NotTo(HaveOccurred())
-			return utils.EnvVarExists(utils.RAY_REDIS_ADDRESS, headPod.Spec.Containers[utils.RayContainerIndex].Env)
-		}
-		return false
-	}, TestTimeoutMedium).Should(BeTrue())
+	g.Eventually(checkEnv(test, namespace.Name, utils.RAY_REDIS_ADDRESS, rayClusterAC),
```

Comment:
Check the environment after the cluster is `rayv1.AllPodRunningAndReadyFirstTime` so that you don’t need to use `Eventually`. The environment should be injected into the PodSpec whenever the head pod is created.